components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -592,7 +592,7 @@ struct SwitchChainRequest {
 interface JsonRpcService {
   // Checks the chain ID for an ethereum chain that should be added
   AddEthereumChain(EthereumChain chain) => (string chain_id, bool accepted);
-  AddEthereumChainForOrigin(EthereumChain chain, url.mojom.Url origin) => (string chain_id, bool accepted);
+  AddEthereumChainForOrigin(EthereumChain chain, url.mojom.Url origin) => (string chain_id, bool accepted, string error_message);
```

Comment:
Could we update this and `AddEthereumChain` while you're here to return a `ProviderError` instead of `bool accepted`? I think you'll need to update some android code, you can probably just let CI check it is ok rather than building on Android.

Reply:
will update this function in https://github.com/brave/brave-browser/issues/20605